src/realm-settings/PartialImport.tsx

```diff
@@ -278,107 +285,258 @@ export const PartialImportDialog = (props: PartialImportProps) => {
+                    setIsCollisionSelectOpen(!isCollisionSelectOpen);
+                  }}
+                  onSelect={handleCollisionSelect}
+                  placeholderText={t(collisionOption)}
                 >
-                  {realmSelectOptions()}
+                  {collisionOptions()}
                 </Select>
               </StackItem>
-            )}
-            <StackItem>
-              <Text>{t("chooseResources")}:</Text>
-              <DataList aria-label="Resources to import" isCompact>
-                {targetHasResource("users") &&
-                  resourceDataListItem("users", "users")}
-                {targetHasResource("groups") &&
-                  resourceDataListItem("groups", "groups")}
-                {targetHasResource("clients") &&
-                  resourceDataListItem("clients", "clients")}
-                {targetHasResource("identityProviders") &&
-                  resourceDataListItem(
-                    "identityProviders",
-                    "identity providers"
-                  )}
-                {targetHasRealmRoles() &&
-                  resourceDataListItem("realmRoles", "realm roles")}
-                {targetHasClientRoles() &&
-                  resourceDataListItem("clientRoles", "client roles")}
-              </DataList>
-            </StackItem>
-            <StackItem>
-              <Text>{t("selectIfResourceExists")}:</Text>
-              <Select
-                isOpen={isCollisionSelectOpen}
-                direction="up"
-                onToggle={() => {
-                  setIsCollisionSelectOpen(!isCollisionSelectOpen);
-                }}
-                onSelect={handleCollisionSelect}
-                placeholderText={t(collisionOption)}
-              >
-                {collisionOptions()}
-              </Select>
-            </StackItem>
-          </>
-        )}
-      </Stack>
-    </Modal>
+            </>
+          )}
+        </Stack>
+      </Modal>
+    );
+  };
+
+  const importCompleteMessage = () => {
+    return t("importSuccess", {
+      added: importResponse?.added,
+      overwritten: importResponse?.overwritten,
+      skipped: importResponse?.skipped,
+    });
+  };
+
+  const loader = async (first?: number, max?: number) => {
+    let last = first! + max!;
+    const length = importResponse?.results.length;
+    if (last > length!) last = length!;
+
+    if (importResponse?.results)
+      return importResponse.results.slice(first, last);
+
+    // Empty state can't happen as you will get back a result record
+    // for each record you submit. This is here to make Typescript happy.
+    return [];
+  };
+
+  const ActionLabel = (importRecord: { action: string }) => {
+    let text = t("added");
+    let color: "green" | "orange" | "purple" = "green";
+
+    if (importRecord.action === "SKIPPED") {
+      text = t("skipped");
+      color = "orange";
+    }
+
+    if (importRecord.action === "OVERWRITTEN") {
```

Comment:
Since these branches are mutually exclusive you might want to use `else if` here.
```suggestion
    } else if (importRecord.action === "OVERWRITTEN") {
```

Reply:
I'll go one better.  How about no if statement at all?
```
  const ActionLabel = (importRecord: { action: string; id: string }) => {
    const actionLabels = new Map([
      [
        "ADDED",
        <Label key={importRecord.id} color="green">
          {t("added")}
        </Label>,
      ],
      [
        "SKIPPED",
        <Label key={importRecord.id} color="orange">
          {t("skipped")}
        </Label>,
      ],
      [
        "OVERWRITTEN",
        <Label key={importRecord.id} color="purple">
          {t("overwritten")}
        </Label>,
      ],
    ]);

    return actionLabels.get(importRecord.action);
  };
```

Reply:
BTW, I went ahead and added the above code.

Reply:
This is a good improvement, but you could take it a step further and remove the need to keep a `Map` around by using a `switch` statement and returning the values directly. This might also be more performant as it does not allocate the extra elements in memory and is likely more easily optimized by the JIT.

```tsx
const ActionLabel = (importRecord: { action: string; id: string }) => {
  switch (importRecord.action) {
    case "ADDED":
      return (
        <Label key={importRecord.id} color="green">
          {t("added")}
        </Label>
      );
    case "SKIPPED":
      return (
        <Label key={importRecord.id} color="green">
          {t("added")}
        </Label>
      );
    case "OVERWRITTEN":
      return (
        <Label key={importRecord.id} color="purple">
          {t("overwritten")}
        </Label>
      );
  }
};
```